README-branching.md

```diff
@@ -0,0 +1,96 @@
+- `o1js-main`: compatible with testnet, but has latest `proof-systems` features so that they can be used in `o1js`
+  - Contains mina changes from `rampup`
+  - But `proof-systems/develop` which by default is used by `mina/develop`.
+  - Uses `o1js/main` and `o1js-bindings/main` as explained [here](https://github.com/o1-labs/o1js/blob/main/README-dev.md#branch-compatibility?).
+  - When `proof-systems/develop` is too cutting-edge and the adaptations of its changes haven't been landed in mina, `o1js` will use the `proof-systems/o1js-main` branch which is lagging behind `proof-systems/develop` a bit.
+
+
+The relationship between the branches is as presented: `master ⊆ compatible ⊆ rampup ⊆ berkeley ⊆ develop`.
+- This means `compatible` includes all the changes in `master`, `rampup` all the changes in `compatible` and so on. So `develop` contains all the changes from *all* the stable branches, but also contains features that do not exist in any of the "subsets".
+- The back-merging direction is thus left-to-right: whenever a feature lands in this chain, it has to be periodically "back-merged" to the right.
+- The branches are merged in the other direction (upstream) only when released.
+- When merely a new feature is introduced, it should aim at the exact target branch. This place depends on the feature, e.g. `compatible` for softfork features, `develop` for more experimental/next release, etc. And then the merged feature is back-propagated downstream (to the right).
+
+
+![Illustration of the branching strategy](docs/res/branching_flow_david_wong.png)
+
+
+
+### Hard forks / releases:
+
+Whenever a hard fork happens, the code in the corresponding release branch, e.g. `berkeley`, is released.  When this happens, the current `compatible` is entirely discarded and a new `compatible` gets created based off of `develop`:
+- `release/1.X.X` branches are made off of `compatible` and tagged with alpha and beta tags until the code is deemed stable, then the `release/1.X.X` branch is merged into `master` and given a stable tag. Whenever code is tagged, if anything is missing in in the downstream branches (compatible, develop) then the tagged branch is also merged back for consistency.
+- So after Berkeley release: `berkeley` branch will become the new `master`. `berkeley` will be removed from `proof-systems`. `develop` will be renamed into `compatible`.
+
+
+## Day to day: which branch should I use?
+
+When developing a feature, use the general description of the branches above to decide. Here's a quick rule:
+- If a feature/enhancement/bug fix is not feature breaking, and scoped for a mainnet then base it off of `compatible`. If you’re not sure whether or not your changes are breaking, they probably are not and should build upon `compatible`.
```

Comment:
shouldn't?

Reply:
No, the intention of this was to say: don't worry, in most cases you can actually go with `compatible` since most changes are not breaking. I believe this statement was a part of the original document in notion...